rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala

```diff
@@ -113,16 +113,13 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
 
   type Result = (Blake2b256Hash, Option[PersistedData], HistoryAction)
 
-  protected[this] val dataLogger: F[Logger] =
-    Sync[F].delay(Logger("coop.rchain.rspace.datametrics"))
+  protected[this] val dataLogger: Logger =
+    Logger("coop.rchain.rspace.datametrics")
 
-  private def measure(actions: List[HotStoreAction]): F[Unit] =
-    for {
-      d <- dataLogger
-      _ = d.whenDebugEnabled {
-        computeMeasure(actions).foreach(p => d.debug(p))
-      }
-    } yield ()
+  private def measure(actions: List[HotStoreAction]): Unit =
```

Comment:
I think it would be prudent to put measure in `Sync[F].delay(`
e.g.
```
Sync[F].delay {
dataLogger.whenDebugEnabled {
  computeMeasure(actions).foreach(p => dataLogger.debug(p))
}
```
as we want to make sure it is executed with the rest of the effect, not on invocation.

Reply:
Definitely, as `measure` isn't pure

Reply:
It doesn't look like this was resolved

Reply:
I had pushed thhttps://github.com/lukasz-golebiewskie wrong branch. Would you plz review again @lukasz-golebiewski ?